cmd/vic-machine/configure/configure.go

```diff
@@ -111,6 +113,12 @@ func (c *Configure) processParams() error {
 		return err
 	}
 
+	var err error
+	usingDHCP := ip.IsUnspecifiedIP(c.Data.ClientNetwork.IP.IP) || ip.IsUnspecifiedIP(c.Data.PublicNetwork.IP.IP) || ip.IsUnspecifiedIP(c.Data.ManagementNetwork.IP.IP)
+	if c.DNS, err = c.dns.ProcessDNSServers(usingDHCP); err != nil {
```

Comment:
In reconfigure, this logic is not incorrect. User might not specify ip during reconfiguration, but that just means they're not trying to change ip settings this time.
If the dns value is set to empty, the default value should be reset after we query back old configuration, and based on that, to see if dhcp is used or not.